pallets/liquid-staking-two/src/lib.rs

```diff
@@ -0,0 +1,318 @@
+
+use frame_support::{
+    pallet_prelude::*, traits::SortedMembers, transactional, BoundedVec, PalletId,
+};
+use frame_system::pallet_prelude::*;
+use orml_traits::XcmTransfer;
+use sp_runtime::{traits::AccountIdConversion, ArithmeticError, FixedPointNumber, RuntimeDebug};
+use sp_std::convert::TryInto;
+use sp_std::prelude::*;
+use xcm::v0::{Junction, MultiLocation, NetworkId};
+
+use orml_traits::{MultiCurrency, MultiCurrencyExtended};
+
+pub use pallet::*;
+use primitives::{Amount, Balance, CurrencyId, EraIndex, LiquidStakingProtocol, ExchangeRateProvider, Rate, Ratio};
+
+#[derive(Copy, Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug)]
+pub enum StakingOperationType {
+	Bond,
+	BondExtra,
+	Unbond,
+	Rebond,
+	TransferToRelaychain,
+	RecordReward,
+	RecordSlash,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug)]
+pub enum Phase {
+	Started,
+	UpdateEraIndex,
+	RecordReward,
+	EmitEventToRelaychain,
+	RecordStakingOperation,
+	Finished,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug)]
+pub enum ResponseStatus {
+	Ready,
+	Processing,
+	Successed,
+	Failed,
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+
+    use super::*;
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// The overarching event type.
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        /// Currency type used for staking and liquid assets
+        type Currency: MultiCurrencyExtended<
+            Self::AccountId,
+            CurrencyId = CurrencyId,
+            Balance = Balance,
+            Amount = Amount,
+        >;
+
+        /// Currency used for staking
+        #[pallet::constant]
+        type StakingCurrency: Get<CurrencyId>;
+
+        /// Currency used for liquid voucher
+        #[pallet::constant]
+        type LiquidCurrency: Get<CurrencyId>;
+
+        /// The pallet id of liquid staking, keeps all the staking assets.
+        #[pallet::constant]
+        type PalletId: Get<PalletId>;
+
+        /// The origin which can withdraw staking assets.
+        type WithdrawOrigin: EnsureOrigin<Self::Origin>;
+
+        /// XCM transfer
+        type XcmTransfer: XcmTransfer<Self::AccountId, Balance, CurrencyId>;
+
+        /// Approved agent list on relaychain
+        type Members: SortedMembers<Self::AccountId>;
+
+        /// Base xcm weight to use for cross chain transfer
+        type BaseXcmWeight: Get<Weight>;
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        /// ExchangeRate is invalid
+        InvalidExchangeRate,
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// The assets get staked successfully
+        Staked(T::AccountId, Balance),
+        /// The voucher get unstaked successfully
```

Comment:
Either `The voucher gets unstaked successfully`

or `The vouchers get unstaked successfully`